From d94fc5ce1dc395747c3934ecffcdec0396583755 Mon Sep 17 00:00:00 2001
From: Nick Hainke <vincent@systemli.org>
Date: Fri, 26 May 2023 19:46:33 +0200
Subject: [PATCH] dmaengine: mediatek: add HSDMA support for mt7621

Commit 87dd67f496f7 ("staging: mt7621-dma: remove driver from tree")
removed the mt7621-dma driver. Move the driver from staging to the
folder "drivers/dma/mediatek" containing already other mediatek dma
driver implementations and maintain it downstream in OpenWrt.

This patch will not be sent to upstream linux. It is just a workaround.

Signed-off-by: Nick Hainke <vincent@systemli.org>
---
 drivers/dma/mediatek/Kconfig  | 6 ++++++
 drivers/dma/mediatek/Makefile | 1 +
 2 files changed, 7 insertions(+)

diff --git a/drivers/dma/mediatek/Kconfig b/drivers/dma/mediatek/Kconfig
index 7a46a5455957..1a24d73ce87f 100644
--- a/drivers/dma/mediatek/Kconfig
+++ b/drivers/dma/mediatek/Kconfig
@@ -36,3 +36,9 @@ config MTK_UART_APDMA
 	  When SERIAL_8250_MT6577 is enabled, and if you want to use DMA,
 	  you can enable the config. The DMA engine can only be used
 	  with MediaTek SoCs.
+
+config MTK_HSDMA
+	tristate "MTK HSDMA support"
+	depends on RALINK && SOC_MT7621
+	select DMA_ENGINE
+	select DMA_VIRTUAL_CHANNELS
diff --git a/drivers/dma/mediatek/Makefile b/drivers/dma/mediatek/Makefile
index 5ba39a5edc13..784af039b7c0 100644
--- a/drivers/dma/mediatek/Makefile
+++ b/drivers/dma/mediatek/Makefile
@@ -2,3 +2,4 @@
 obj-$(CONFIG_MTK_UART_APDMA) += mtk-uart-apdma.o
 obj-$(CONFIG_MTK_HSDMA) += mtk-hsdma.o
 obj-$(CONFIG_MTK_CQDMA) += mtk-cqdma.o
+obj-$(CONFIG_MTK_HSDMA) += hsdma-mt7621.o
-- 
2.40.1

